### DISCUSSION TITLE
rootful installation

### DISCUSSION THREAD

**Discussion by craigeous** [thatmarcel/beepserv-rewrite#13] [Q&A]
I'm running an iPhone 6s iOS 15.8.1 with a rootful palera1n jailbreak. when i go to install the rootful version of beepserv i get an error about needing mobilesubstrate installed. i installed cydia substrate, but seem to get the same error. just wondering if anyone has gotten this installed/setup with a rootful jailbreak.

i'm trying rootful because with rootless my phone was losing the jailbreak at least once a day and it was just a hung pain to keep it going.

**Comment by thatmarcel** [comment]
That's odd.
While rootful palera1n is not recommended anymore and therefore not officially supported by this project, you could try installing `libhooker` or `ellekit` to provide `mobilesubstrate`.

Also note that you'll need to install [oldabi](https://github.com/evelyneee/OldABI) (from the [ElleKit repo](https://repo.ellekit.space)) to use the rootful version on iOS 15+.
